src/tests/Microsoft.PowerFx.Core.Tests/BindingEngineTests.cs

```diff
@@ -199,6 +204,49 @@ public void TableInRegressionError()
             AssertContainsError(result, "Error 0-7: Invalid argument type. Cannot use Table values in this context.");
         }
 
+        [Fact]
+        public void CheckParseResultSideEffects()
+        {
+            var config = new PowerFxConfig();
+            config.AddFunction(new DummyFunction());
+
+            var fncs = config.Functions.Where(fnc => !fnc.IsSelfContained).ToList();
+            var engine = new Engine(config);
+            var formula = "Dummy(); Dummy()";
+            var options = new ParserOptions { AllowsSideEffects = true };
+
+            var result1 = engine.Check(formula, options: options);
+            Assert.True(result1.IsSuccess);
+
+            var parseResult2 = engine.Parse(formula, options);
+            var result2 = engine.Check(parseResult2, options: options);
+            Assert.True(result2.IsSuccess);
+        }
+
+        private class DummyFunction : TexlFunction

```

Comment:
We need a function with behavior in the test above. However, I couldn't find any public function that has behavior and `Set` is defined in interpreter tests.

Reply:
This approach is great, but Can we rename to "BehaviorFunction" or add a comment so that we know why we're defining a new function here.

Reply:
Did that